06-Template-Distribution-Protocol.md

```diff
@@ -65,12 +65,22 @@ The primary template-providing function. Note that the `coinbase_tx_outputs` byt
 | coinbase_tx_outputs_count   | U32            | The number of transaction outputs included in coinbase_tx_outputs     |
 +-----------------------------+----------------+-----------------------------------------------------------------------+
 | coinbase_tx_outputs         | B0_64K         | Bitcoin transaction outputs to be included as the last outputs in     |
-|                             |                | the coinbase transaction                                              |
+|                             |                | the coinbase transaction. An additional coinbase tx output (other than|
+|                             |                | the output which pays the fees + block subsidy), will most likely be  |
+|                             |                | the coinbase witness commitment output.                               |
+|                             |                |                                                                       |
 +-----------------------------+----------------+-----------------------------------------------------------------------+
 | coinbase_tx_locktime        | U32            | The locktime field in the coinbase transaction                        |
 +-----------------------------+----------------+-----------------------------------------------------------------------+
 | merkle_path                 | SEQ0_255[U256] | Merkle path hashes ordered from deepest                               |
 +-----------------------------+----------------+-----------------------------------------------------------------------+
+| witness_reserve_value       | B0_255         | Optional value when a NewTemplate contains segwit inputs. The witness |
```

Comment:
I don't understand the purpose of this field.

Why would the client side need to the know witness_reserve**d**_value?
Will it ever recalculate the witness commitment hash?

As far as I understand the client has a freedom to add additional output to the coinbase transaction with the maximum size of `coinbase_output_max_additional_size`. Nothing more. The structure of the block, witness data, and specifically, the witness commitment is fixed by the bitcoin core and cant be manipulated. Sending the w_reserved_value to the client then has no purpuse. Or if the client was ever to recalculate witness commitment hash. It would also need the entire witness tree.

https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#commitment-structure

Reply:
So, the most simple case is:

server will provide  coinbase_tx_outputs (an output that pay 0 and contain the witness commitment hash).

client will create the coinbase using <client-specified-outputs> || coinbase_tx_outputs adding the output/s that pay the pool, so witness commitment will be the output with the highest index.

There are also cases where server want to add more than one output and in that case we need to use coinbase_tx_value_remaining.


Not sure why we need witness_reserve_value @ccdle12 ?